packages/ui/src/components/popover/popover.tsx

```diff
@@ -0,0 +1,212 @@
+import {h, Component} from 'preact';
+import * as styles from './popover.scss';
+
+export enum PopoverVerticalPositions {
+  Top = 'top',
+  Bottom = 'bottom',
+}
+export enum PopoverHorizontalPositions {
+  Left = 'left',
+  Right = 'right',
+}
+export enum PopoverTriggerMode {
+  Click = 'click',
+  Hover = 'hover',
+}
+
+export enum KeyboardKeys {
+  Esc = 27,
+  Enter = 13,
+}
+
+const CLOSE_ON_HOVER_DELAY = 500;
+
+const defaultProps = {
+  verticalPosition: PopoverVerticalPositions.Top,
+  horizontalPosition: PopoverHorizontalPositions.Left,
+  triggerMode: PopoverTriggerMode.Click,
+  className: 'popover',
+  closeOnEsc: true,
+  closeOnClick: true,
+};
+
+interface PopoverProps {
+  onClose?: () => void;
+  onOpen?: () => void;
+  closeOnClick: boolean;
+  closeOnEsc: boolean;
+  verticalPosition: PopoverVerticalPositions;
+  horizontalPosition: PopoverHorizontalPositions;
+  className: string;
+  triggerMode: PopoverTriggerMode;
+  content: JSX.Element;
+  children: JSX.Element | JSX.Element[];
+}
+
+interface PopoverState {
+  open: boolean;
+}
+
+/**
+ * Popover renders popup with a target.
+ * Properties description:
+ *   onOpen - function that will be executed when popover opens;
+ *   onClose - function that will be executed when popover closes;
+ *   closeOnClick - close the popover on mouse click;
+ *   verticalPosition - vertical position of popover relative to target ("top" or "bottom"), default - "top";
+ *   horizontalPosition - horizontal position of popover relative to target ("left" or "right"), default - "left";
+ *   className - popover class, can be use for popover styling, default - 'popover';
+ *   closeOnEsc - handle ESC keyboard pressed event and close popover, default - true;
+ *   triggerMode - popover support 2 ways for opening: click (keyPress) and mouse hover ("click" or "hover"), default - "click";
+ *   content - content of popover. Can be any valid Preact node, ex:
+ *      <select>
+ *          <option>Option 1</option>
+ *          ...
+ *          <option>Option n</option>
+ *      </select>
+ *   children - popover target. Can be any valid Preact node, ex:
+ *      <button>
+ *          <i className="icon" />
+ *      </button>
+ */
+
+export class Popover extends Component<PopoverProps, PopoverState> {
```

Comment:
please add componentWillUnmount and call `this._removeListeners();
` to prevent memory leaks